backend/users/auth.py

```diff
@@ -10,17 +10,59 @@
 
 from .models import LoginGovUser
 
+from audit.models import Access
+
+import logging
 import string
 import random
 
+logger = logging.getLogger(__name__)
+
+
+def claim_audit_access(user, all_emails):
+    access_invites = (
+        Access.objects.filter(user_id=None)
+        .filter(email__in=all_emails)
+        .update(user_id=user.id)
+    )
+    logger.debug(f"{user.email} granted access to {access_invites} new audits")
+
 
 class JWTUpsertAuthentication(JWTAuthentication):
+    def get_or_create_auth_user(self, email, all_emails):
+        # find all Users where email is in LoginGov all_emails
+        users = self.user_model.objects.filter(email__in=all_emails)
+
+        if users.count() > 1:
+            # more than one found, take the one that matches the LoginGov primary email
+            try:
+                user = users.get(email=email)
```

Comment:
Is it possible that there might be more than one user with the same email address?

Reply:
per our Slack discussion: uniqueness of `User` emails is not enforced at the database level, but this scenario should not be possible at the application level, given the way new `Users` are created. It is possible, though, if `Users` are created manually, either via direct database insertions or through Django admin. So, we explicitly catch and log that failure mode and ultimately reject the authentication attempt